### DISCUSSION TITLE
Question about otSrpClientSetHostName()

### DISCUSSION THREAD

**Discussion by OlivierGre** [openthread/openthread#8181] [Q&A] [answered]
I have a question concerning this code in SrpClient::Process():

https://github.com/openthread/openthread/blob/24d633d99d1e336ea7db0bdb62b2961e4d110a40/src/cli/cli_srp_client.cpp#L134-L149

In the second part of this code, why do we have to retrieve a pointer to the persisted butter, memcpy the data to it and then call otSrpClientSetHostName() ?
This is strange to me because it is like if the update of the host name was done by the memcpy, that means outside otSrpClientSetHostName().

Thanks for your information

**Comment by jwhui** [comment, marked as answer]
The SRP Client only maintains a pointer to the host name provided by the caller. As a result, the caller must maintain the buffer while the host name is in use. As the comment mentions, the first call to `otSrpClientSetHostName()` simply checks whether we can set the host name successfully.

https://github.com/openthread/openthread/blob/0756d1ea8fee17eba0cdee42a34ea353413a4f99/src/cli/cli_srp_client.cpp#L140-L144

However, the first call will set the pointer to a buffer that the client cannot maintain (i.e. `aArgs[1].GetCString()`).

https://github.com/openthread/openthread/blob/0756d1ea8fee17eba0cdee42a34ea353413a4f99/src/cli/cli_srp_client.cpp#L146

The `memcpy()` copies that data into a buffer that the client can maintain and the second call to `otSrpClientSetHostName()` updates the SRP Client with that new buffer (i.e. `hostName`).

https://github.com/openthread/openthread/blob/0756d1ea8fee17eba0cdee42a34ea353413a4f99/src/cli/cli_srp_client.cpp#L149

**Reply by OlivierGre** [reply]
Ok thank you Jonathan
